src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -697,10 +715,15 @@ def __write_status_file(self):
             shutil.rmtree(self.complete_status_file_path)
 
         # Write complete status file <seq.no>.complete.status
-        self.env_layer.file_system.write_with_retry_using_temp_file(self.complete_status_file_path, '[{0}]'.format(json.dumps(complete_status_payload)), mode='w+')
+        status_file_payload_json_dumps = json.dumps(complete_status_payload)
+        self.env_layer.file_system.write_with_retry_using_temp_file(self.complete_status_file_path, '[{0}]'.format(status_file_payload_json_dumps), mode='w+')
 
-        # Write agent status file
-        self.env_layer.file_system.write_with_retry_using_temp_file(self.status_file_path, '[{0}]'.format(json.dumps(complete_status_payload)), mode='w+')
+        if Constants.StatusTruncationConfig.TURN_ON_TRUNCATION:
+            self.composite_logger.log_debug("Perform truncation on status file if applicable")
```

Comment:
log_verbose
We do not want excessive logging into telemetry as is slows down execution

Reply:
pending for review because I've modified it in multiple areas